include/pybind11/pybind11.h

```diff
@@ -1193,29 +1193,27 @@ class class_ : public detail::generic_type {
         }
     }
 
+    static void init_holder_from_existing(instance_type *inst, const holder_type *holder_ptr,
+                                          std::true_type /*is_copy_constructible*/) {
+        new (&inst->holder) holder_type(*holder_ptr);
+    }
+
+    static void init_holder_from_existing(instance_type *inst, const holder_type *holder_ptr,
+                                          std::false_type /*is_copy_constructible*/) {
+        new (&inst->holder) holder_type(std::move(*const_cast<holder_type *>(holder_ptr)));
+    }
```

Comment:
I don't think another layer of abstraction is needed here since this is all implementation detail and the intent should be clear from `init_holder_helper` right below.